src/Cake.Core/IO/FileExtensions.cs

```diff
@@ -100,5 +100,49 @@ public static IEnumerable<string> ReadLines(this IFile file, System.Text.Encodin
                 return result;
             }
         }
+
+        /// <summary>
+        /// Checks if file has CLR PE Header.
+        /// </summary>
+        /// <param name="file">The file to be read from.</param>
+        /// <returns><c>true</c> if file is CLR assembly; otherwise, <c>false</c>.</returns>
+        public static bool IsCLRAssembly(this IFile file)
```

Comment:
@devlead I don't doubt that this function does exactly what it says, but is it possible to provide a link to some sort of reference as to what exactly is going on here?

Reply:
We've been using this method for awhile to aid with tool resolution. I just moved it and made it a public extension method.

Essentially most windows dll:s/exe:s have pe headers, a little can be read link below or in win SDK.

https://docs.microsoft.com/en-us/windows/desktop/debug/pe-format

Reply:
@devlead I spotted that after I wrote the comment as I got to the bottom of the changes.  That URL seems very useful.  Can we get it added into the remarks section of the XML comment?

Reply:
sure.